### PR TITLE
Typo fixes & clarity improvements

### PR DESCRIPTION
**Pull request by divarvel**
Along with typo fixes, I tried to add explanations everywhere i found things hard to understand

additions I think would be useful

- describing the different fields in the `Biscuit` display (explaining that `authority` is the first block, repeating that each block can declare new facts, symbols (with specific restrictions), etc)
- describing a bit more syntax elements (`@`, `matches`, etc). Is the operations list exhaustive in the types list?
- showing in one place the verifier input (the facts and the rules are listed separately, maybe displaying all the input elements in one place, like for the biscuit itself, would make things clearer?)
- mentioning that variables also introduce symbols

The rule of the last caveat is presented without a leading `*`, is this intentional?

also, are there comments defined in the datalog constraint?

### CODE REVIEW

**Comment by Geal** [comment]
the operations list should be exhaustive.
More info on the inputs, state of the veriier, etc would be good too.
The missing `*` was not intentional (although I'm considering removing that point of syntax, which is confusing)
There are no comments defined in the datalog syntax, but maybe I should add some?

**Comment by divarvel** [comment]
Comments would be useful at least for the examples here. By the end of the document, the interaction between `right`, `resource` and the caveat rules get harder to follow.

For the operations list, are `@` and `matches` supposed to be in it, or are they different concepts?

**Comment by divarvel** [comment]
I have added a description of the contents of a biscuit.

There are two questions left:

- what about the scoping of symbols? if a block sees the scope of previous blocks, is shadowing allowed?
- what is `context` ?

File: SUMMARY.md [hunk 1]
```diff
@@ -107,16 +110,30 @@ ancestor("Bob", "Denise")
...
+ancestor("Bob", "Charles")
+ancestor("Charles", "Denise")
+ancestor("Alice", "Charles")
+ancestor("Bob", "Denise")
+ancestor("Alice", "Denise")
+```
+
+Interactions with a Datalog program are done through queries: **a query contains
+a rule** that we apply over the system, and **it returns the generated facts**.
+>>>>>>> 94e52e1... Typo fixes & clarity improvements
```
Comment (Geal) [inline SUMMARY.md:129]: leftover from a merge?

File: SUMMARY.md [hunk 2]
```diff
@@ -148,19 +168,31 @@ using a string constraint:
...
 
+## `#ambient` and `#authority` symbols
+
+This caveat uses a _symbol_ named `#ambient` (symbols start with a `#`).
+
+There are two special symbols that can appear in facts:
+
+-`#ambient`: facts that are _provided by the verifier_, and that depend on the **request**, like which resource we want to access(file path, REST endpoint, etc), operation(read, write...), current date and time, source IP address, HTTP headers...
+- `#authority`: facts _defined by the token's original creator_ or _the verifier_, that indicates the basic rights of the **token**. Every new attenation of the token will reduce those rights by adding caveats
+>>>>>>> 94e52e1... Typo fixes & clarity improvements
```
Comment (Geal) [inline SUMMARY.md:188]: here too

File: SUMMARY.md [hunk 3]
```diff
@@ -186,9 +218,23 @@ Biscuit {
...
+ - `blocks` carries a list of blocks, which can refine the scope of the macaroon.
+
+Here, `authority` provides the initial block, which can be refined in subsequent blocks.
+
 A block comes with new symbols it adds to the system (there's a default symbol
 table that already contains values like `#authority` or `#operation`). It can
-contain facts, rules and caveats.
+contain facts, rules and caveats. A block contains:
+
+ - `symbols` a block can introduce new symbols (TODO are those symbols usable in following blocks?)
```
Comment (Geal) [inline SUMMARY.md:233]: symbols added in a block are sable in the following blocks, and the library makes sure a symbol is not declared twice when deserializing

File: SUMMARY.md [hunk 4]
```diff
@@ -186,9 +218,23 @@ Biscuit {
...
+ - `context` TODO
```
Comment (Geal) [inline SUMMARY.md:234]: `context` is a free text field, to add comments or help in querying data on the backend

File: SUMMARY.md [hunk 5]
```diff
@@ -186,9 +218,23 @@ Biscuit {
...
+ - `facts` each block can define new facts (but only only `authority` can define facts mentioning `#authority`)
+ - `rules` each block can define new rules
```
Comment (Geal) [inline SUMMARY.md:236]: only the `authority` block can define rules that generate facts with `#authority`

File: SUMMARY.md [hunk 6]
```diff
@@ -186,9 +218,23 @@ Biscuit {
...
+Here, `authority` provides the initial block, which can be refined in subsequent blocks.
+
 A block comes with new symbols it adds to the system (there's a default symbol
 table that already contains values like `#authority` or `#operation`). It can
-contain facts, rules and caveats.
+contain facts, rules and caveats. A block contains:
+
+ - `symbols` a block can introduce new symbols (TODO are those symbols usable in following blocks?)
+ - `context` TODO
+ - `facts` each block can define new facts (but only only `authority` can define facts mentioning `#authority`)
```
Comment (Geal) [inline SUMMARY.md:235]: "only only"

**Review by Geal — CHANGES_REQUESTED** [review]

File: SUMMARY.md (hunk 3 above)
Comment (divarvel) [reply SUMMARY.md]: in this case, isn't the `symbols` part of the biscuit equivalent to the symbols part of the `authority` block?

**Comment by divarvel** [comment]
I have fixed the issues, and removed the todos (by fixing them).

File: SUMMARY.md (hunk 3 above)
Comment (Geal) [reply SUMMARY.md]: the symbols for the whole token also include the symbol table used by the token creator and the verifier. They need to use the same one, and it contains some often used values. Here the authority symbols list only contains `"user_id"`, the rest comes from the symbol table

**Pull request merged**